packages/wellknown-did-verifier/__tests__/restAgent.test.ts

```diff
@@ -0,0 +1,114 @@
+import 'cross-fetch/polyfill'
+// @ts-ignore
+import express from 'express'
+import { Server } from 'http'
+import { IAgent, createAgent, IAgentOptions } from '@veramo/core'
+import { AgentRestClient } from '@veramo/remote-client'
+import { AgentRouter, RequestWithAgentRouter } from '@veramo/remote-server'
+import { getConfig } from '@veramo/cli/build/setup'
+import { createObjects } from '@veramo/cli/build/lib/objectCreator'
+import { IWellKnownDidVerifier } from '../src/types/IWellKnownDidVerifier';
+import { ServiceTypesEnum } from '@sphereon/wellknown-dids-client/dist/types'
+import wellKnownDidVerifierAgentLogic from './shared/wellKnownDidVerifierAgentLogic'
+
+jest.setTimeout(30000)
+
+const port = 3002
+const basePath = '/agent'
+
+let serverAgent: IAgent
+let restServer: Server
+
+const getAgent = (options?: IAgentOptions) =>
+  createAgent<IWellKnownDidVerifier>({
+    ...options,
+    plugins: [
+      new AgentRestClient({
+        url: 'http://localhost:' + port + basePath,
+        enabledMethods: serverAgent.availableMethods(),
+        schema: serverAgent.getSchema(),
+      }),
+    ],
+  })
+
+const setup = async (): Promise<boolean> => {
+  const config = getConfig('packages/wellknown-did-verifier/agent.yml')
+  const { agent } = createObjects(config, { agent: '/agent' })
+
+  await agent.registerSignatureVerification({
+    key: 'verified',
+    signatureVerification: () => Promise.resolve({ verified: true })
+  }, null)
+
+  const DID = 'did:key:z6MkoTHsgNNrby8JzCNQ1iRLyW5QQ6R8Xuu6AA8igGrMVPUM';
+  const ORIGIN = 'https://example.com';
+  const DOCUMENT = {
+    '@context': [
+        'https://www.w3.org/ns/did/v1',
+      'https://identity.foundation/.well-known/did-configuration/v1'
+    ],
+    id: DID,
+    verificationMethod: [
+      {
+        id: `${DID}#_Qq0UL2Fq651Q0Fjd6TvnYE-faHiOpRlPVQcY_-tA4A`,
+        type: 'JsonWebKey2020',
+        controller: DID,
+        publicKeyJwk: {
+          kty: 'OKP',
+          crv: 'Ed25519',
+          x: 'VCpo2LMLhn6iWku8MKvSLg2ZAoC-nlOyPVQaO3FxVeQ',
+        },
+      },
+    ],
+    service: [
+      {
+        id: `${DID}#foo`,
+        type: ServiceTypesEnum.LINKED_DOMAINS,
+        serviceEndpoint: {
+          origins: [ORIGIN, ORIGIN],
+        },
+      },
+      {
+        id: `${DID}#bar`,
+        type: ServiceTypesEnum.LINKED_DOMAINS,
+        serviceEndpoint: ORIGIN,
+      },
+    ],
+  };
+
+  agent.resolveDid = jest.fn().mockReturnValue(Promise.resolve({ didDocument: DOCUMENT }))
```

Comment:
Isn't this exactly the same as in the setup() method? (The Document and mock)

Reply:
not sure if i understand the question. so the reason why this is a bity weird is because i had to mock the context of the agent. i tried mocking it directly in the tests but that did not work. so had to duplicate it for both local and rest tests

Reply:
Yeah I meant the setup() method in the localAgent. Not a biggie